methods.go

```diff
@@ -2,10 +2,34 @@ package telego
 
 import (
 	"fmt"
+	"math"
+	"time"
 
 	"github.com/mymmrac/telego/telegoapi"
 )
 
+// `retry` - executes a function multiple times and returns the final `err` value.
+// It finishes when either `err == nil` or `maxAttempts` is reached.
+func (b *Bot) retry(fn func() error) error {
```

Comment:
I would suggest implementing this functionality inside `performRequest` method, there is no need to repeat this N number of times for each method

Reply:
Basically, retry in my mind should work for each Telegram call, if user wants to implement retry only for `SendMessage` for example, it can be done outside of lib

The only thing is that `GetUpdates` method is somewhat special and Telego already has retry logic for it, so we should not retry it as all other methods

Reply:
Correct, `Caller` provides a way to call Telegram (as you sad, abstraction above HTTP clients), `performRequest` gathers all needed info about method call (all parameters & method URL) and using `Caller` does the request. This is very low level things in Telego with which users rarely should do something.

As of what I suggested, there is no need to change each method (like `SendMessage`, `SendPhoto`, `SendVideo`, and so on), all of them are using `performRequest` so you can add retry logic inside `performRequest`